packages/wrangler/src/config.ts

```diff
@@ -1,131 +1,479 @@
+   * You might have more than one account, so make sure to use
+   * the ID of the account associated with the zone/route you
+   * provide, if you provide one. It can also be specified through
+   * the CF_ACCOUNT_ID environment variable.
+   *
+   * @optional
+   * @inherited
+   */
+  account_id?: string;
 
-type Dev = {
-  ip?: string;
-  port?: number;
-  local_protocol?: string;
-  upstream_protocol?: string;
-};
+  /**
+   * The project "type". A holdover from wrangler 1.x.
+   * Valid values were "webpack", "javascript", and "rust".
+   *
+   * @deprecated DO NOT USE THIS.
+   * @optional
+   * @inherited
+   * @breaking
+   */
+  type?: "webpack" | "javascript" | "rust";
 
-export type Vars = { [key: string]: string };
+  /**
+   * A date in the form yyyy/mm/dd, which will be used to determine
+   * which version of the Workers runtime is used. More details at
+   * https://developers.cloudflare.com/workers/platform/compatibility-dates
+   * @optional for `dev`, @required for `publish`
+   * @inherited
+   */
+  compatibility_date?: string;
 
-type Cron = string; // TODO: we should be able to parse a cron pattern with ts
+  /**
+   * A list of flags that enable features from upcoming features of
+   * the Workers runtime, usually used together with compatibility_flags.
+   * More details at
+   * https://developers.cloudflare.com/workers/platform/compatibility-dates
+   *
+   * @optional
+   * @inherited
+   * @todo This could be an enum!
+   */
+  compatibility_flags?: string[];
 
-type KVNamespace = {
-  binding: string;
-  preview_id?: string;
-  id: string;
-};
+  /**
+   * Whether we use <name>.<subdomain>.workers.dev to
+   * test and deploy your worker.
+   *
+   * @default `true` (This is a breaking change from wrangler 1)
+   * @optional
+   * @inherited
+   * @breaking
+   */
+  workers_dev?: boolean;
 
-type DurableObject = {
-  name: string;
-  class_name: string;
-  script_name?: string;
-};
+  /**
+   * The zone ID of the zone you want to deploy to. You can find this
+   * in your domain page on the dashboard.
+   *
+   * @deprecated This is unnecessary since we can deduce this from routes directly
+   * @optional
+   * @inherited
+   */
+  zone_id?: string;
 
-type Service = {
-  name: string;
-  service: string;
-  environment: string;
-};
+  /**
+   * A list of routes that your worker should be deployed to.
+   * Only one of `routes` or `route` is required.
+   *
+   * @required when workers_dev is false
+   * @inherited
+   */
+  routes?: string[];
+
+  /**
+   * A route that your worker should be deployed to. Literally
+   * the same as routes, but only one.
+   * Only one of `routes` or `route` is required.
+   *
+   * @required when workers_dev is false
+   * @inherited
+   */
+  route?: string;
+
+  /**
+   * Path to the webpack config to use when building your worker.
+   * A holdover from wrangler 1.x, used with `type: "webpack"`.
+   *
+   * @deprecated DO NOT USE THIS.
```

Comment:
I think it would be helpful for these `@deprecated` comments to give a short blurb of what to do instead; or at least why we decided to remove it.